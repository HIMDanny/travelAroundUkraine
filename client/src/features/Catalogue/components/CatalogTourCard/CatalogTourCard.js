import React from 'react';
import { styled, Stack, CardActions, CardMedia, Button, Typography, CardContent, Box } from '@mui/material';
import AccessTimeIcon from '@mui/icons-material/AccessTime';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import FavoriteBorderSharpIcon from '@mui/icons-material/FavoriteBorderSharp';
import FavoriteSharpIcon from '@mui/icons-material/FavoriteSharp';
import { useDispatch, useSelector } from 'react-redux';
import { addItemtoWishList, deleteItemfromWishList } from '../../../../store/slices/inFavoritesSlice/inFavoritesSlice';
import { ReactComponent as CoinsIcon } from '../../../../assets/svg/CoinsIcon.svg';

const CardContainer = styled(Stack)(({ theme }) => ({
  backgroundColor: theme.palette.primary.contrastText,
  borderRadius: 10,
  boxShadow: '0px 0px 40px rgba(0, 0, 0, 0.05)',
  maxWidth: '1000px',
}));

const CardImage = styled(CardMedia)(({ theme }) => ({
  width: '100%',
  height: 240,
  borderRadius: '10px 10px 0 0',

  [theme.breakpoints.up('tablet')]: {
    width: 250,
    height: 'auto',
    borderRadius: '10px 0 0 10px',
  },
}));

const CardTitle = styled((props) => <Typography variant="h3" {...props} />)(({ theme }) => ({
  color: theme.palette.text.primary,
  marginTop: 20,
  marginBottom: 20,
  textTransform: 'uppercase',
}));

const TourDescription = styled(Typography)(({ theme }) => ({
  borderLeft: `3px solid ${theme.palette.primary.main}`,
  paddingLeft: 24,
  height: '50px',
  overflow: 'hidden',
  WebkitLineClamp: '3',
  display: '-webkit-box',
  WebkitBoxOrient: 'vertical',
}));

const CardButton = styled(Button)(({ theme }) => ({
  padding: '12px 85px',
  marginTop: 15,

  [theme.breakpoints.up('tablet')]: {
    padding: '12px 43px',
  },
}));

const CatalogTourCard = ({
  name,
  currentPrice,
  duration,
  description,
  imageUrls,
  id,
  inFavorites,
  itemNo,
  inFavoritesCounter,
}) => {
  const dispatch = useDispatch();
  const isLogin = useSelector((store) => store.user.isLogin);
  const navigate = useNavigate();

  return (
    <CardContainer direction={{ xs: 'column', tablet: 'row' }} spacing={1}>
      <CardImage component="img" image={imageUrls[0]} alt="tour photo" />
      <CardContent sx={{ padding: '0 30px 0 36px' }}>
        <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
          <CardTitle>{name}</CardTitle>
          {inFavorites === false ? (
            <FavoriteBorderSharpIcon onClick={() => dispatch(addItemtoWishList(isLogin, id))} />
          ) : (
            <FavoriteSharpIcon onClick={() => dispatch(deleteItemfromWishList(isLogin, id, inFavoritesCounter))} />
          )}
        </Box>
        <Stack direction="row" spacing={3} alignItems="center" sx={{ mb: '20px' }}>
          <Stack direction="row" spacing={1} alignItems="center">
            <CoinsIcon />
            <Typography variant="h3" component="span">
              {currentPrice} â‚¬
            </Typography>
          </Stack>

          <Stack direction="row" spacing={1} alignItems="center">
            <AccessTimeIcon color="primary" sx={{ height: '21px' }} />
            <Typography variant="h3" component="span">
              {duration}
            </Typography>
          </Stack>
        </Stack>
        <TourDescription>{description}</TourDescription>
        <CardActions>
          <CardButton onClick={() => navigate(`/tour/${+itemNo}`)}>More details</CardButton>
        </CardActions>
      </CardContent>
    </CardContainer>
  );
};

CatalogTourCard.propTypes = {
  name: PropTypes.string.isRequired,
  currentPrice: PropTypes.number.isRequired,
  duration: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  imageUrls: PropTypes.arrayOf(PropTypes.string).isRequired,
  itemNo: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  inFavorites: PropTypes.bool.isRequired,
  inFavoritesCounter: PropTypes.number,
};

CatalogTourCard.defaultProps = {
  inFavoritesCounter: 0,
};

export default CatalogTourCard;
