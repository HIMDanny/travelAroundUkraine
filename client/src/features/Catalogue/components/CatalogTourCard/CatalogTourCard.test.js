import { render } from '@testing-library/react';
import CatalogTourCard from './CatalogTourCard';

jest.mock('@mui/material/Stack', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/CardActions', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/CardMedia', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/CardContent', () => ({ children }) => <div>{children}</div>);

describe('CatalogTourCard snapshot test', () => {
  test('should CatalogTourCard match snapshot', () => {
    const { asFragment } = render(
      <CatalogTourCard
        name="ItemName"
        description="Itemdescription"
        currentPrice={1}
        duration="ItemDuration"
        imageUrls={['ItemImage']}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
