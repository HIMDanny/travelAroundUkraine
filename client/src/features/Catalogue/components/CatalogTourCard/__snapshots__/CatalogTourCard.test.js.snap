// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogTourCard snapshot test should CatalogTourCard match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <div>
      <p>
        ItemName
      </p>
      <div>
        <div>
          <svg>
            CoinsIcon.svg
          </svg>
          <p>
            1 â‚¬
          </p>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1fb58gc-MuiSvgIcon-root"
            data-testid="AccessTimeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          <p>
            ItemDuration
          </p>
        </div>
      </div>
      <p>
        Itemdescription
      </p>
      <div>
        <button>
          More details
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
